src/libraries/System.IO.FileSystem.AccessControl/tests/TempAclDirectory.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Security.AccessControl;
+using System.Security.Principal;
+
+namespace System.IO
+{
+    /// <summary>
+    /// Represents a temporary directory.
+    /// Disposing will recurse all files and directories inside it, ensure the
+    /// appropriate access control is set, then delete all of them.
+    /// </summary>
+    public sealed class TempAclDirectory : TempDirectory
+    {
+        protected override void DeleteDirectory()
+        {
+            try
+            {
+                var identity = new SecurityIdentifier(WellKnownSidType.BuiltinUsersSid, null);
+                var accessRule = new FileSystemAccessRule(identity, FileSystemRights.FullControl, AccessControlType.Allow);
```

Comment:
why `accessRule` is not used anywhere?

Reply:
My mistake. It should've been passed to the DirectorySecurity or FileSecurity object by invoking AddAccessRule. I fixed this, re-ran the unit tests locally (with the try catch commented) and they are passing (no files/folders are failing to get deleted).